tests/test_dwarf.py

```diff
@@ -1680,6 +1680,83 @@ def test_template_value_parameter_missing_value(self):
+                    ),
+                    (
+                        DwarfDie(
+                            DW_TAG.GNU_template_parameter_pack,
+                            (DwarfAttrib(DW_AT.name, DW_FORM.string, "Params"),),
+                            (
+                                DwarfDie(
+                                    DW_TAG.template_type_parameter,
+                                    (
+                                        DwarfAttrib(
+                                            DW_AT.type, DW_FORM.ref4, "int_die"
+                                        ),
+                                        DwarfAttrib(DW_AT.name, DW_FORM.string, "T"),
+                                    ),
+                                ),
+                                DwarfDie(
+                                    DW_TAG.template_value_parameter,
+                                    (
+                                        DwarfAttrib(
+                                            DW_AT.type, DW_FORM.ref4, "unsigned_int_die"
+                                        ),
+                                        DwarfAttrib(DW_AT.name, DW_FORM.string, "N"),
+                                        DwarfAttrib(
+                                            DW_AT.const_value, DW_FORM.data1, 2
+                                        ),
+                                    ),
+                                ),
+                            ),
+                        ),
+                    ),
+                ),
+                *labeled_int_die,
+                *labeled_unsigned_int_die,
+            )
+        )
+        self.assertIdentical(
+            prog.type("TEST").type,
+            prog.class_type(
+                "ParamPack<int, 123>",
+                template_parameters=(
+                    TypeTemplateParameter(prog.int_type("int", 4, True), "T"),
+                    TypeTemplateParameter(
+                        Object(prog, prog.int_type("unsigned int", 4, False), 2), "N"
+                    ),
+                ),
+            ),
+        )
+
+    def test_invalid_parameter_pack(self):
+        with self.assertRaisesRegex(
+            Exception, "template parameter has invalid DW_TAG: 36"
+        ):
+            dwarf_program(
+                wrap_test_type_dies(
+                    DwarfDie(
+                        DW_TAG.class_type,
+                        (DwarfAttrib(DW_AT.declaration, DW_FORM.flag_present, True),),
+                        (
+                            DwarfDie(
+                                DW_TAG.GNU_template_parameter_pack,
+                                (DwarfAttrib(DW_AT.name, DW_FORM.string, "Params"),),
+                                (char_die,),
+                            ),
+                        ),
+                    ),
+                    *labeled_unsigned_int_die,
+                )
+            ).type("TEST").type.template_parameters[0].argument
```

Comment:
I don't think we should require that all of the children of `DW_TAG_GNU_template_parameter_pack` are either `DW_TAG_template_type_parameter` or `DW_TAG_template_value_parameter`. It's better to be forwards-compatible and allow producers to stick anything else they want in there. (This means that the parsing code needs to handle this differently, too.)